force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,84 @@
+/**
+ * @description       :
+ * @author            : Adrienne D. Millican
+ * @group             :
+ * @last modified on  : 06-22-2024
+ * @last modified by  : ChangeMeIn@UserSettingsUnder.SFDoc
+ **/
+
+trigger OpportunityTrigger on Opportunity(
+  before update,
+  after update,
+  after delete
+) {
+  if (Trigger.isBefore) {
+    if (Trigger.isUpdate && !TriggerHandler.hasRunBeforeUpdate) {
+      TriggerHandler.hasRunBeforeUpdate = true;
+      for (Opportunity opp : Trigger.newMap.values()) {
+        if (opp.Amount <= 5000) {
+          opp.addError('Opportunity amount must be greater than 5000');
+        }
+      }
+    }
+  }
+
+  if (Trigger.isAfter) {
+    if (Trigger.isUpdate && !TriggerHandler.hasRunAfterUpdate) {
+      TriggerHandler.hasRunAfterUpdate = true;
+      Set<Id> acctIdSet = new Set<Id>();
+      List<Opportunity> oppsToUpdLst = new List<Opportunity>();
+      for (Opportunity opp : Trigger.newMap.values()) {
+        if (opp.AccountId != null) {
+          acctIdSet.add(opp.AccountId);
+        }
+      }
+      List<Contact> contactsOnAcctLst = [
+        SELECT Id, AccountId
+        FROM Contact
+        WHERE AccountId IN :acctIdSet AND Title = 'CEO'
+      ];
+
+      for (Opportunity opp : [
+        SELECT Id, AccountId
+        FROM Opportunity
+        WHERE Id IN :Trigger.newMap.keyset()
```

Comment:
You should not need to query the accountid from the opportunity. You should already have it in your trigger.new opps